examples/http-server/main.go

```diff
@@ -27,7 +31,12 @@ func main() {
 	a.GET("/mysql", MysqlHandler)
 
 	// Run the application
-	a.Run()
+	go a.Run()
```

Comment:
I would have refactored to pass the context to `a.Run` so the context could be used when calling data sources, and not only in the shutdown method

But that would be a huge refactoring that may change framework signature. Maybe it's what you were afraid of.

Maybe to be addressed in another PR/issue

Reply:
Probably signal handling and cancellation could be done inside `app.Run()` 
It would keep users from having to know that they have to implement handling shutdown by themselves, thus simplified examples and no reason to explain it in docs